### PR TITLE
Add unit tests to validate references in KB

### PR DESCRIPTION
**Pull request by Zakaria-Latif**
References validation test for meta.json files by caching URL checks to avoid redundant requests and to spot the invalid URLs

### CODE REVIEW

File: MOBILE_CLIENT/IOS/_IMPORTANT/IPA_ENCRYPTED/meta.json [hunk 1]
```diff
@@ -2,9 +2,6 @@
   "risk_rating": "important",
   "short_description": "Application is encrypted by Apple, limiting analysis capabilities.",
   "references": {
-    "Decryption apps guide": "https://kov4l3nko.github.io/blog/2016-03-01-decrypting-apps-from-appstore/",
-    "Repackaging applications guide": "https://labs.mwrinfosecurity.com/blog/repacking-and-resigning-ios-applications/",
-    "lldb-dumpdecrypted": "https://github.com/BlueCocoa/dumpdecrypted-lldb",
```
Comment (adnaneserrar) [inline MOBILE_CLIENT/IOS/_IMPORTANT/IPA_ENCRYPTED/meta.json:7]: why deleting those?

Comment (Zakaria-Latif) [reply MOBILE_CLIENT/IOS/_IMPORTANT/IPA_ENCRYPTED/meta.json:7]: they are dead

File: MOBILE_CLIENT/ANDROID/_HIGH/APK_WIFI_API_PII/meta.json [hunk 2]
```diff
@@ -3,8 +3,7 @@
     "short_description": "Android WiFi API exposes Personal Identifiable Information (PII), potentially allowing apps with appropriate permissions to access sensitive information through WiFi-related APIs on Android devices.",
     "references": {
         "Android Docs": "https://source.android.com/docs/security/bulletin/2015-12-01#information_disclosure_vulnerability_in_wi-fi",
-        "NVD": "https://nvd.nist.gov/vuln/detail/CVE-2015-6629",
-        "Vuldb": "https://vuldb.com/?id.221977"
```
Comment (mohsinenar) [inline MOBILE_CLIENT/ANDROID/_HIGH/APK_WIFI_API_PII/meta.json:7]: This link is still valid!

File: MOBILE_CLIENT/COMMON/_MEDIUM/INSECURE_PASSWORD_STORAGE/meta.json [hunk 3]
```diff
@@ -6,7 +6,6 @@
     "OWASP Insecure Cryptographic Storage": "https://www.owasp.org/index.php/Top_10_2010-A7-Insecure_Cryptographic_Storage",
     "OWASP Insecure Storage": "https://www.owasp.org/index.php/Insecure_Storage",
     "CWE-522 Insufficiently Protected Credentials": "https://cwe.mitre.org/data/definitions/522.html",
-    "CWE-312 Cleartext Storage of Sensitive Information": "CWE-312: Cleartext Storage of Sensitive Information",
```
Comment (mohsinenar) [inline MOBILE_CLIENT/COMMON/_MEDIUM/INSECURE_PASSWORD_STORAGE/meta.json:9]: https://cwe.mitre.org/data/definitions/312.html

File: MOBILE_CLIENT/IOS/_IMPORTANT/IPA_ENCRYPTED/meta.json (hunk 1 above)
Comment (mohsinenar) [reply MOBILE_CLIENT/IOS/_IMPORTANT/IPA_ENCRYPTED/meta.json:7]: @Zakaria-Latif Try to find other references, cause now we have None!

**Review by mohsinenar — DISMISSED** [review]

File: MOBILE_CLIENT/COMMON/_POTENTIALLY/SECRETS_REVIEW/meta.json [hunk 4]
```diff
@@ -3,7 +3,6 @@
   "short_description": "Passwords, Tokens or other sensitive information are stored in the application.",
   "references": {
     "API Security Best Practices": "https://developers.google.com/maps/api-security-best-practices",
-    "How to secure your Google Maps API Key": "https://www.storemapper.com/support/knowledge-base/how-to-secure-your-google-maps-api-key/",
```
Comment (ohachimOs) [inline MOBILE_CLIENT/COMMON/_POTENTIALLY/SECRETS_REVIEW/meta.json:6]: I don't understand why this change?

File: MOBILE_CLIENT/ANDROID/_HARDENING/APK_FACEBOOK_REACT_DEV_SETTINGS_ENABLED/meta.json [hunk 5]
```diff
@@ -2,7 +2,6 @@
   "risk_rating": "hardening",
   "short_description": "Application is compiled with `DevSettingsActivity` developer activity exposed.",
   "references": {
-    "Facebook SDK Debug Logs": "https://www.appsloveworld.com/react-native/100/11/react-native-whats-com-facebook-react-devsupport-devsettingsactivity"
```
Comment (amine3) [inline MOBILE_CLIENT/ANDROID/_HARDENING/APK_FACEBOOK_REACT_DEV_SETTINGS_ENABLED/meta.json:5]: If some links are invalid, please try to find valid ones for the same topic.

**Review by amine3 — DISMISSED** [review]

File: MOBILE_CLIENT/COMMON/_POTENTIALLY/SECRETS_REVIEW/meta.json (hunk 4 above)
Comment (Zakaria-Latif) [reply MOBILE_CLIENT/COMMON/_POTENTIALLY/SECRETS_REVIEW/meta.json:6]: Some links are broken, causing the unittest to fail

**Review by 3asm — APPROVED** [review]

**Pull request merged**